### PR TITLE
Add interface identifier conversion

### PR DESCRIPTION
**Pull request by faern**
Adding bindings to functions allowing converting from network interface aliases, names, GUIDs and indexes to their `NET_LUID` representation, and back again.

### CODE REVIEW

File: src/shared/netioapi.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright © 2015-2017 winapi-rs developers
+// Licensed under the Apache License, Version 2.0
+// <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// All files in the project carrying such notice may not be copied, modified, or distributed
+// except according to those terms.
+use shared::basetsd::SIZE_T;
+use shared::guiddef::GUID;
+use shared::ifdef::{NET_IFINDEX,NET_LUID,PNET_IFINDEX,PNET_LUID};
```
Comment (retep998) [inline src/shared/netioapi.rs]: There should be a space after the comma.

File: src/shared/netioapi.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+use shared::minwindef::DWORD;
+use shared::ntdef::{PWSTR,WCHAR};
```
Comment (retep998) [inline src/shared/netioapi.rs]: Here as well.

File: src/shared/ifdef.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// Copyright © 2015-2017 winapi-rs developers
```
Comment (retep998) [inline src/shared/ifdef.rs]: Shouldn't the copyright year be just 2018?

File: src/shared/netioapi.rs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+// Copyright © 2015-2017 winapi-rs developers
```
Comment (retep998) [inline src/shared/netioapi.rs]: Here as well.

**Review by retep998 — CHANGES_REQUESTED** [review]

File: src/shared/netioapi.rs (hunk 4 above)
Comment (faern) [reply src/shared/netioapi.rs]: Fixed

File: src/shared/ifdef.rs (hunk 3 above)
Comment (faern) [reply src/shared/ifdef.rs]: Fixed

File: src/shared/netioapi.rs (hunk 2 above)
Comment (faern) [reply src/shared/netioapi.rs]: Fixed

File: src/shared/netioapi.rs (hunk 1 above)
Comment (faern) [reply src/shared/netioapi.rs]: Fixed

**Comment by faern** [comment]
I added the `A` version where there were `W` versions present. Also squashed everything to one commit and added the PR number to the commit message, as seems to be the standard here.

**Comment by retep998** [comment]
The adding of the PR number to the commit message is done by the Squash and merge feature of Github which I use to merge PRs.

**Comment by faern** [comment]
All feedback has been addressed. So I think this is ready for merge?

**Review by retep998 — APPROVED** [review]

**Pull request merged**